doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,812 @@
+    }
+    FindPackagesOptions CreateFindOptions()
+    {
+        com_ptr<abi::IFindPackagesOptions> abifindOptions = nullptr;
+        check_hresult(CoCreateInstance(CLSID_FindPackagesOptions, nullptr, CLSCTX_LOCAL_SERVER, IID_PPV_ARGS(abifindOptions.put())));
+        check_pointer(abifindOptions.get());
+        FindPackagesOptions findPackagesOptions{ abifindOptions.as<FindPackagesOptions>() };
+        return findPackagesOptions;
+    } 
+    GetCompositeAppCatalogOptions CreateGetCompositeAppCatalogOptions()
+    {
+        com_ptr<abi::IGetCompositeAppCatalogOptions> abiGetCompositeAppCatalogOptions = nullptr;
+        check_hresult(CoCreateInstance(CLSID_GetCompositeAppCatalogOptions), nullptr, CLSCTX_LOCAL_SERVER, IID_PPV_ARGS(abiGetCompositeAppCatalogOptions.put())));
+        check_pointer(abiGetCompositeAppCatalogOptions.get());
+        GetCompositeAppCatalogOptions getCompositeAppCatalogOptions{ abiGetCompositeAppCatalogOptions.as<GetCompositeAppCatalogOptions>() };
+        return getCompositeAppCatalogOptions;
+    }
+```
+
+## 3.2. Search
+
+The api can be used to search for packages in a catalog known to Windows Package Manager. This can be used to get availability information
+or start an install.
+
+```c++ (C++ish pseudocode)
+
+
+    IAsyncOperation<Package> FindPackageInCatalog(AppCatalog catalog, std::wstring packageId)
+    {
+        FindPackagesOptions findPackagesOptions = CreateFindOptions();
+        PackageMatchFilter filter;
+        filter.IsAdditive = true;
+        filter.Field = PackageMatchField::Id;
+        filter.Type = MatchType::Exact;
+        filter.Value = packageId;
+        findPackagesOptions.Filters().Append(filter);
+        FindPackagesResult findPackagesResult{ co_await catalog.FindPackagesAsync(findPackagesOptions) };
+
+        winrt::IVectorView<ResultMatch> matches = findPackagesResult.Matches();
+        co_return matches.GetAt(0).Package();
+    }
+
+    IAsyncOperation<Package> MainPage::FindPackage()
+    {
+        // Capture the ui thread context.
+        winrt::apartment_context uiThread;
+        co_await winrt::resume_background();
+
+        AppInstaller appInstaller = CreateAppInstaller();
+        AppCatalog catalog{ co_await appInstaller.GetAppCatalogAsync(PredefinedAppCatalog::OpenWindowsCatalog) };
+        co_await catalog.OpenAsync();
+        Package package = { co_await FindPackageInCatalog(catalog, m_installAppId) };
+
+        co_return package;
+    }
+
+```
+
+## 3.3. Install
+
+```c++ (C++ish pseudocode)
+
+    IAsyncOperationWithProgress<InstallResult, InstallProgress> InstallPackage(Package package)
+    {
+        AppInstaller appInstaller = CreateAppInstaller();
+        InstallOptions installOptions = CreateInstallOptions();
+
+        installOptions.InstallScope(InstallScope::User);
+        installOptions.Package(package);
+        installOptions.InstallMode(InstallMode::Silent);
+
+        return appInstaller.InstallPackageAsync(installOptions);

```

Comment:
`co_return` here

Reply:
Is it? This is intended to synchronously return the IAsyncOperation from the InstallPackageAsync call.